plugins/experimental/stale_while_revalidate/stale_while_revalidate.c

```diff
@@ -145,17 +105,8 @@ create_request_info(TSHttpTxn txn)
     TSHttpHdrClone(req_info->buf, buf, loc, &(req_info->http_hdr_loc));
     TSHandleMLocRelease(buf, TS_NULL_MLOC, loc);
 
-    sa = TSHttpTxnClientAddrGet(txn);
-    switch (sa->sa_family) {
-    case AF_INET:
-      memcpy(&req_info->client_addr.sin, sa, sizeof(struct sockaddr_in));
-      break;
-    case AF_INET6:
-      memcpy(&req_info->client_addr.sin6, sa, sizeof(struct sockaddr_in6));
-      break;
-    default:
-      break;
-    }
+    // src is IpEndpoint (union w/struct sockaddr)
+    memmove(&req_info->client_addr, TSHttpTxnClientAddrGet(txn), sizeof(struct sockaddr));
```

Comment:
```
(gdb) p sizeof(struct sockaddr)
$1 = 16
(gdb) p sizeof(struct sockaddr_in)
$2 = 16
(gdb) p sizeof(struct sockaddr_in6)
$3 = 28
```

Please revert the sockaddr change.

Reply:
That is unsafe as is.   Coulda sworn it was 32 bytes .. but it's not.   We need big enough space for everything.   Just as with the others ... more mallocs and pointers will only make freeing harder and less atomic and higher fragments.

Reply:
Many systems have sockaddr_len ... and I'm wondering why this doesn't.